python/fledge/services/core/api/auth.py

```diff
@@ -253,42 +256,64 @@ async def create_user(request):
     data = await request.json()
     username = data.get('username')
     password = data.get('password')
+    certificate = data.get('certificate')
     role_id = data.get('role_id', DEFAULT_ROLE_ID)
 
-    if not username or not password:
-        _logger.warning("Username and password are required to create user")
-        raise web.HTTPBadRequest(reason="Username or password is missing")
+    if not username:
+        msg = "Username is required to create user"
+        _logger.warning(msg)
+        raise web.HTTPBadRequest(reason=msg, body=json.dumps({"message": msg}))
 
-    if not isinstance(password, str):
-        _logger.warning(PASSWORD_ERROR_MSG)
-        raise web.HTTPBadRequest(reason=PASSWORD_ERROR_MSG)
+    if not (password or certificate):
+        msg = "Either password or certificate is required to create user"
+        _logger.warning(msg)
+        raise web.HTTPBadRequest(reason=msg, body=json.dumps({"message": msg}))
 
-    if not re.match(PASSWORD_REGEX_PATTERN, password):
-        _logger.warning(PASSWORD_ERROR_MSG)
-        raise web.HTTPBadRequest(reason=PASSWORD_ERROR_MSG)
+    if password:
+        if not isinstance(password, str):
+            _logger.warning(PASSWORD_ERROR_MSG)
+            raise web.HTTPBadRequest(reason=PASSWORD_ERROR_MSG, body=json.dumps({"message": PASSWORD_ERROR_MSG}))
+
+        if not re.match(PASSWORD_REGEX_PATTERN, password):
+            _logger.warning(PASSWORD_ERROR_MSG)
+            raise web.HTTPBadRequest(reason=PASSWORD_ERROR_MSG, body=json.dumps({"message": PASSWORD_ERROR_MSG}))
+
+    if certificate:
+        valid_extensions = ('.cert', '.cer', '.crt', '.pem')
+        if not certificate.endswith(valid_extensions):
+            msg = "Accepted cert extensions are {}".format(valid_extensions)
+            _logger.warning(msg)
+            raise web.HTTPBadRequest(reason=msg, body=json.dumps({"message": msg}))
+        is_cert_found = await User.Objects.find_cert(certificate)
+        if not is_cert_found:
+            msg = "{} is not found in certificate store".format(certificate)
+            _logger.warning(msg)
+            raise web.HTTPBadRequest(reason=msg, body=json.dumps({"message": msg}))
```

Comment:
Why a certificate required to reside in fledge. User should get the cert from issuing authority and issuing authority CA cert is the only thing we need in fledge cert store to verify the issue certificate.

Reply:
As per specs I have added this check, @MarkRiddoch Could you please confirm?